🛠 Repo: Replace fs-extra with newer fs built-ins

### Tooling Suggestion Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have pulled the latest `main` branch of the repository.
- [x] I have read and agree to Mocha's [Code of Conduct](https://github.com/mochajs/mocha/blob/main/.github/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/mochajs/mocha/blob/main/.github/CONTRIBUTING.md)
- [x] I have searched for [related issues](https://github.com/mochajs/mocha/issues?q=is%3Aissue) and [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20), but none matched my issue.
- [x] I want to provide a PR to resolve this

### Overview

I ran [`eslint-plugin-depend`](https://github.com/es-tooling/eslint-plugin-depend) on the Mocha repo. It had two reports about `fs-extra`:

```md
/Users/josh/repos/mocha/test/integration/helpers.js
  5:12  error  "fs-extra" should be replaced with an alternative package. Read more here: https://github.com/es-tooling/module-replacements/blob/main/docs/modules/fs-extra.md  depend/ban-dependencies

/Users/josh/repos/mocha/test/integration/options/watch.spec.js
  3:12  error  "fs-extra" should be replaced with an alternative package. Read more here: https://github.com/es-tooling/module-replacements/blob/main/docs/modules/fs-extra.md  depend/ban-dependencies
```

Per https://github.com/es-tooling/module-replacements/blob/main/docs/modules/fs-extra.md:

> `fs-extra` adds some file system methods that previously were not included in the native `fs` module and added `Promise` support to the `fs` methods.
> Many of the fs-extra modules have since been implemented natively by Node.

Proposal: let's switch to native `node:fs` / `node:fs/promises`.

### Additional Info

Note that both are within `test/integration`. This shouldn't impact users.

--- a/.github/workflows/mocha.yml
+++ b/.github/workflows/mocha.yml
@@ -66,7 +66,7 @@ jobs:
     with:
       os: 'ubuntu-latest,windows-latest'
       # The 22.11.0 is instead of 22 per https://github.com/mochajs/mocha/issues/5278
-      node-versions: '14,16,18,20,22.11.0'
+      node-versions: '18,20,22.11.0'
       npm-script: test-node:${{ matrix.test-part }}
       coverage: ${{ matrix.coverage }}
 

--- a/package.json
+++ b/package.json
@@ -129,7 +129,6 @@
     "cross-env": "^7.0.2",
     "eslint": "^8.56.0",
     "fail-on-errors-webpack-plugin": "^3.0.0",
-    "fs-extra": "^10.0.0",
     "globals": "^13.24.0",
     "installed-check": "^9.3.0",
     "jsdoc": "^3.6.7",

--- a/test/integration/helpers.js
+++ b/test/integration/helpers.js
@@ -2,7 +2,8 @@
 
 const escapeRegExp = require('escape-string-regexp');
 const os = require('os');
-const fs = require('fs-extra');
+const fs = require('fs');
+const fsP = require('fs/promises');
 const {format} = require('util');
 const path = require('path');
 const Base = require('../../lib/reporters/base');
@@ -487,7 +488,7 @@ const touchRef = new Date();
  * @param {string} filepath - Path to file
  */
 function touchFile(filepath) {
-  fs.ensureDirSync(path.dirname(filepath));
+  fs.mkdirSync(path.dirname(filepath), { recursive: true });
   try {
     fs.utimesSync(filepath, touchRef, touchRef);
   } catch (e) {
@@ -519,8 +520,8 @@ function replaceFileContents(filepath, pattern, replacement) {
  */
 function copyFixture(fixtureName, dest) {
   const fixtureSource = resolveFixturePath(fixtureName);
-  fs.ensureDirSync(path.dirname(dest));
-  fs.copySync(fixtureSource, dest);
+  fs.mkdirSync(path.dirname(dest), { recursive: true });
+  fs.cpSync(fixtureSource, dest);
 }
 
 /**
@@ -528,12 +529,12 @@ function copyFixture(fixtureName, dest) {
  * @returns {Promise<CreateTempDirResult>} Temp dir path and cleanup function
  */
 const createTempDir = async () => {
-  const dirpath = await fs.mkdtemp(path.join(os.tmpdir(), 'mocha-'));
+  const dirpath = await fsP.mkdtemp(path.join(os.tmpdir(), 'mocha-'));
   return {
     dirpath,
     removeTempDir: async () => {
       if (!process.env.MOCHA_TEST_KEEP_TEMP_DIRS) {
-        return fs.remove(dirpath);
+        return fs.rmSync(dirpath, { recursive: true, force: true });
       }
     }
   };

--- a/test/integration/options/watch.spec.js
+++ b/test/integration/options/watch.spec.js
@@ -1,6 +1,6 @@
 'use strict';
 
-const fs = require('fs-extra');
+const fs = require('fs');
 const path = require('path');
 const {
   copyFixture,
@@ -131,7 +131,7 @@ describe('--watch', function () {
         [testFile, '--watch-files', 'lib/**/*.xyz'],
         tempDir,
         () => {
-          fs.removeSync(watchedFile);
+          fs.rmSync(watchedFile, { recursive: true, force: true });
         }
       ).then(results => {
         expect(results, 'to have length', 2);

